src/base/RadioCheckboxBase.js

```diff
@@ -0,0 +1,137 @@
+import React, { Component, PropTypes } from 'react'
+import Radium from 'radium'
+
+const NOOP = () => {} // eslint-disable-line no-empty-function
+
+const STYLES = {
+  button: {
+    margin: '0 10px -7px 0',
+    backgroundSize: 'contain',
+    backgroundRepeat: 'no-repeat',
+    backgroundPosition: 'center',
+    backgroundColor: 'transparent',
+    appearance: 'none',
+    '-ms-appearance': 'none',
+    '-moz-appearance': 'none',
+    '-webkit-appearance': 'none',
+    width: 22,
+    height: 22,
+  },
+  wrapEl: {
+    display: 'flex',
+  }
+}
+
+function btnBkgStyle(asset) {
+  return {
+    backgroundImage: `url(${asset})`,
+    ':active': {
+      backgroundImage: `url(${asset})`,
+      border: '0 solid transparent'
+    }
+  }
+}
+
+class RadioCheckboxBase extends Component {
+  static propTypes = {
+    aria          : PropTypes.shape({
+      label         :PropTypes.string,
+    }),
+    assets        : PropTypes.object,
+    btnType       : PropTypes.oneOf(['radio', 'checkbox']).isRequired,
+    children      : PropTypes.string,
+    id            : PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+    isChecked     : PropTypes.bool,
+    onClick       : PropTypes.func,
+    styles        : PropTypes.shape({
+      button        : PropTypes.object,
+      label         : PropTypes.object,
+      wrapEl        : PropTypes.object,
+    }),
+    value         : PropTypes.string,
+    wrapEl        : PropTypes.string,
+  }
+
+  static defaultProps = {
+    aria        : {},
+    isChecked   : false,
+    onClick     : NOOP,
+    styles      : {},
+    wrapEl      : 'div',
+  }
+
+  state = {
+    isChecked   : this.props.isChecked,
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const { isChecked } = nextProps
+    if (isChecked !== this.props.isChecked) { this.setState({isChecked}) }
+  }
+
+  getStyles() {
+    const { styles, assets } = this.props
+    const { isChecked } = this.state
+    const { btnBkg } = assets
```

Comment:
Is assets always going to be defined as an object? If so, we need to make it a required prop or give it a default prop value

Reply:
Good point. Updating.